### PR TITLE
misc: Upgrade to Kotlin 2.1.0

### PR DESCRIPTION
**Pull request by lauzadis**
*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: build-plugins/build-support/src/main/kotlin/aws/sdk/kotlin/gradle/plugins/artifactsizemetrics/ArtifactSizeMetricsPlugin.kt [hunk 1]
```diff
@@ -38,12 +38,11 @@ class ArtifactSizeMetricsPlugin : Plugin<Project> {
...
-    tasks.register<CollectArtifactSizeMetrics>("artifactSizeMetrics") {
-        group = TASK_GROUP
-        onlyIf { tasks.findByName("jvmJar") != null }
-        dependsOn(tasks.withType<Jar>())
-    }
+private fun Project.subprojectArtifactSizeMetricsTask(): TaskProvider<CollectArtifactSizeMetrics> = tasks.register<CollectArtifactSizeMetrics>("artifactSizeMetrics") {
+    group = TASK_GROUP
+    onlyIf { tasks.findByName("jvmJar") != null }
+    dependsOn(tasks.withType<Jar>())
+}
```
Comment (ianbotsf) [inline build-plugins/build-support/src/main/kotlin/aws/sdk/kotlin/gradle/plugins/artifactsizemetrics/ArtifactSizeMetricsPlugin.kt:45]: **Question**: I'm noticing a lot of reformatting which lengthens lines past 120 chars. Was this an automated reformat or did you manually make these changes for some reason?

File: build-plugins/smithy-build/src/main/kotlin/aws/sdk/kotlin/gradle/codegen/SmithyBuildPlugin.kt [hunk 2]
```diff
@@ -102,8 +101,6 @@ class SmithyBuildPlugin : Plugin<Project> {
         tasks.register<SmithyBuildTask>(TASK_GENERATE_SMITHY_PROJECTIONS) {
             group = "codegen"
             dependsOn(generateSmithyBuild)
-            resolvedCliClasspath.set(codegenConfig)
-            runtimeClasspath.set(codegenConfig)
             buildClasspath.set(codegenConfig)
             smithyBuildConfigs.set(project.files(generateSmithyBuild))
         }
```
Comment (ianbotsf) [inline build-plugins/smithy-build/src/main/kotlin/aws/sdk/kotlin/gradle/codegen/SmithyBuildPlugin.kt:106]: **Question**: Why were these removed?

**Review by ianbotsf — APPROVED** [review]

File: build-plugins/build-support/src/main/kotlin/aws/sdk/kotlin/gradle/plugins/artifactsizemetrics/ArtifactSizeMetricsPlugin.kt (hunk 1 above)
Comment (lauzadis) [reply build-plugins/build-support/src/main/kotlin/aws/sdk/kotlin/gradle/plugins/artifactsizemetrics/ArtifactSizeMetricsPlugin.kt:45]: This was an automated format from ktlint
